### ISSUE TITLE
Deploy step is ignored

### ISSUE THREAD

**Issue by sobolevn** [sobolevn/git-secret#120]
Right now deploy step is ignored. We used to rely on `$GITSECRET_DIST`:
- https://github.com/sobolevn/git-secret/blob/master/.travis.yml#L108
- https://github.com/sobolevn/git-secret/blob/master/.ci/before_deploy.sh#L5

New package is not uploaded:
- https://bintray.com/sobolevn/deb/git-secret/view
- https://bintray.com/sobolevn/rpm/git-secret

**Comment by hurricanehrndz** [comment]
@sobolevn 

What do you want to distribute on? Git-test or gnupg2 or gnupg1?

**Comment by sobolevn** [comment]
I want to distribute `git-secret` as `rpm`, `deb`, and possible `apk` packages. Yeap, you nailed it in the last commit.

**Comment by hurricanehrndz** [comment]
With apk, it really should get submitted into the the Alpine repos, no
reason it shouldn't be. I will submit into the openSUSE repos.

**Comment by nictrix** [comment]
@hurricanehrndz @sobolevn Is there anything I can assist with or is 0.2.3 ready to be released just have to push the packages out?

**Comment by sobolevn** [comment]
It is ready, we just need to push the packages.

@nictrix you can really help if you would fix the current release process. It used to be automated.

**Comment by joshrabinowitz** [comment]
@nictrix @sobolevn @hurricanehrndz 

can someone look at the current CI outputs and detail what is left to fix?

**Comment by joshrabinowitz** [comment]
Related: I see this error at the end of 
https://travis-ci.org/sobolevn/git-secret/jobs/368148728
    Skipping a deployment with the bintray provider because a custom condition was not met

**Comment by sobolevn** [comment]
@joshrabinowitz yeap, you got it!

**Comment by joshrabinowitz** [comment]
Deploy is still not working but we get different errors now:
 - on all builds marked GITSECRET_DIST="deb" or GITSECRET_DIST="rpm"
 
````
/home/travis/.rvm/gems/ruby-2.2.7/gems/dpl-bintray-1.9.5/lib/dpl/provider/bintray.rb:45:in `read': No such file or directory @ rb_sysopen - build/deb_descriptor.json (Errno::ENOENT)
	from /home/travis/.rvm/gems/ruby-2.2.7/gems/dpl-bintray-1.9.5/lib/dpl/provider/bintray.rb:45:in `read_descriptor'
````

**Comment by joshrabinowitz** [comment]
@sobolevn guidance welcome

**Comment by sobolevn** [comment]
I can see that deploy had been ignored before #161 was merged.

> Skipping a deployment with the bintray provider because a custom condition was not met
> Skipping a deployment with the bintray provider because a custom condition was not met

Link: https://travis-ci.org/sobolevn/git-secret/jobs/368148723

But now we face another issue: deploy script can not find `build/deb_descriptor.json`. This file is used to specify metadata for deploy. https://travis-ci.org/sobolevn/git-secret/jobs/368494269#L2439

It should be generated during `before_deploy` script. By `.ci/before_deploy.sh` file. By this particular line: https://github.com/sobolevn/git-secret/blob/master/.ci/before_deploy.sh#L10

Before deploy actually happens `travis` runs a cleanup process. https://travis-ci.org/sobolevn/git-secret/jobs/368494269#L2417 Which also may be the cause why our descriptor is missing.

Well, we have to change the way app is deployed in several things:
1. We need to run deploy on `tags`, right now we are using deploys on `master` branch
2. There should be only two deploys: one for `deb` and one for `rpm`, right now due to `$GITSECRET_DIST == rpm` condition deploys will happen several times: `KITCHEN_REGEXP="gnupg2-centos-latest"; GITSECRET_DIST="rpm"` and `env: KITCHEN_REGEXP="gnupg2-fedora-latest"; GITSECRET_DIST="rpm"` will release two identical builds

We can also check that deploy is working locally. I guess, you will need to register a test account in `bintray`. There are several things to configure: your username, password and `gpg` key for signing releases

**Comment by simbo1905** [comment]
I have setup a pull requestion to fix this which is https://github.com/sobolevn/git-secret/pull/184

**Comment by simbo1905** [comment]
This issue has been fixed with the deployment of 0.2.4 

There is a note by @sobolevn deploying from tags that I will break out as a new ticket.
